kv/README.md

```diff
@@ -0,0 +1,84 @@
+# Proposed design for JetStream based KV services
+
+This is a proposal to design KV services ontop of JetStream, this requires a KV client to be written, the basic
+features are:
+
+ * Multiple named buckets full of keys with n historical values kept per value
+ * Put and Get `string(k)=string(v)` values
+ * Deleting a key completely
+ * Per key TTL
+ * Compacting a key down to n remaining values
+ * Watching a specific key or the entire bucket for live updates
+ * Encoders and Decoders that transforms both keys and values
+ * A read-cache that builds an in-memory cache for fast reads
+ * Ability to read from regional read replicas maintained by JS Mirroring (planned)
+ * read-after-write safety unless the read cache of read replicas were used
+ * Valid keys are `\A[-/_a-zA-Z0-9]+\z` after encoding
```

Comment:
I would allow non wild-carded subjects. just like publish

Reply:
For the Origin stuff - its just cluster now, I dropped Server and user details.  Clients dont know the domain without having to go make extra calls to APIs which we dont want.  Cluster is in the connect/info head.

For the valid key behaviour, I'd prefer to stick to a basic constrained value, much easier to reason about. People can always use `/` as a seperator which is quite common in kv usage